compiler/test/compilable/cppmangle.d

```diff
@@ -528,7 +528,7 @@ version (CppMangle_Itanium)
     static assert(basic_ostream!(char, char_traits!char).food.mangleof == "_ZNSo4foodEv");
     static assert(basic_iostream!(char, char_traits!char).fooe.mangleof == "_ZNSd4fooeEv");
 
-    static assert(func_18957_2.mangleof == `_Z12func_18957_2PSaIiE`);
+    //static assert(func_18957_2.mangleof == `_Z12func_18957_2PSaIiE`);
```

Comment:
`func_18957_2` is an overload set with a normal function and a function template. Up until now dmd just chose the function because of the bug in `isUnique`. I would argue that this should be an error. Thoughts?

Reply:
Perhaps put this comment in the test, so it's clear why the assert is disabled

Reply:
The comment was used as a discussion starter. I don't expect this PR will be merged with this test commented.

Reply:
One way to fix this without introducing any breaking changes is to add an optional bool parameter to `isUnique` which is set only when isUnique is called from `semanticTiargs`. That way, we can preserve the existing behavior with respect to the mangling of functions. Although, I still tend to argue that the way this patch deals with things is the more correct approach.

Reply:
I'm not terribly concerned with changing the name mangling. D libraries compiled with one version of DMD and linking with modules compiled with another version are a risky business regardless.